src/util/http/websocket/WebSocketSession.cpp

```diff
@@ -33,9 +33,20 @@ net::awaitable<void> WebSocketSession::handleClientCommands() {
 
   while (ws_.is_open()) {
     co_await ws_.async_read(buffer, net::use_awaitable);
-    // TODO<RobinTF> replace with cancellation process
-    ws_.text(ws_.got_text());
-    co_await ws_.async_write(buffer.data(), net::use_awaitable);
+    if (ws_.got_text()) {
+      auto data = buffer.data();
+      if (std::string_view{static_cast<char*>(data.data()), data.size()} ==
+          "cancel") {
+        if (auto cancellationHandle =
+                queryRegistry_.getCancellationHandle(queryId_)) {
+          cancellationHandle->cancel(CancellationState::MANUAL);
+
+          co_await ws_.async_close(beast::websocket::close_code::normal,
+                                   net::use_awaitable);
+          break;
+        }
+      }
+    }
```

Comment:
Question: Does a cancelled query send something to the websocket listeners?

Reply:
Only indirectly: If the query has been successfully cancelled, the websocket is closed by the server.